ethers-core/src/types/transaction/response.rs

```diff
@@ -173,6 +191,20 @@ impl Transaction {
                 if let Some(chain_id) = self.chain_id {
                     rlp.append(&normalize_v(self.v.as_u64(), U64::from(chain_id.as_u64())));
                 }
+                rlp.append(&self.r);
+                rlp.append(&self.s);
+            }
+            // Optimism Deposited Transaction
+            #[cfg(feature = "optimism")]
+            Some(x) if x == U64::from(0x7E) => {
+                rlp.append(&self.source_hash.expect("sourceHash is required for deposited tx"));
```

Comment:
should `source_hash` be `<_>` instead of `Option<_>`

Reply:
oh i thought we were in a TypedTransaction context. I suppose this is fine here, but you should use `rlp_opt`, not a panic

you'll also need to add a TypedTransaction enum variant

Reply:
> you'll also need to add a TypedTransaction enum variant

Ok, @prestwich regarding this do you think we should reuse the `Eip1559TransactionRequest` type and extend it with the optimism feature, or instead create a new `OptimismDepositedTransactionRequest` type?

Reply:
if optimism supports both separately, do both